### PR TITLE
FOR-24: Create HomePage View (Work In Progress)

### PR DESCRIPTION
**Pull request by Karththigan**
# Creating a basic Home Page View

Creating a basic home page view that we can link after the Log In and Sign Up views

Note the use of JSwing's `CardLayout`: this lets us go from the Home Page view (picture 1) to the Post View (picture 2) within the same window, without opening a new window. I found this [helpful](https://www.mathcs.duq.edu/simon/Java6/uiswing/layout/card.html).

## TODOs
- [ ] Separate code that populates the home page with posts (The `contentArea `JPanel)
- [ ] Add code to the left navigation bar (one with category buttons) to enable filtering (this was in our UI sketch we submitted in Week 6)
- [ ] Make View Manager Model to update home page with new posts
- [ ] Add limit to how many posts are shown on the home page at a time (currently a for loop populates the home page `contentArea` Panel with dummy posts)
- [ ] Separate Post View
- [ ] Remove "View Post" button and replace with functionality to open a post view upon clicking any part of the `postBox `JPanel
- [ ] Add functionality to display actual posts from DB

### Home Page View
image

### Post View
image

Jira Ticket: [FOR-24](https://cscgroup76.atlassian.net/browse/FOR-24)

[FOR-24]: https://cscgroup76.atlassian.net/browse/FOR-24?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Karththigan** [comment]
## Modified HomePage to match Login view style
image
## Modified Post View to match Login view style
image


## Login View (for reference)
image

**Comment by Karththigan** [comment]
## Added drop down menu to profile button
image

I don't think we are doing a settings page (unless we want to)? But just put "Settings" as a placeholder for now

Also I will move over all the panel switching to acontroller/main later, this is just proof of concept

**Review by ethanliu24 — COMMENTED** [review]
You mentioned that you can switch to post view with CardLayout without opening another window. However I think that should be done through a controller that calls a switch view method in the use case

File: src/main/java/controller/homepage/HomepageState.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package controller.homepage;
+
+/**
+ * Represents the state of the HomePage view.
+ */
+public class HomepageState {
+    private String[] posts;
```
Comment (ethanliu24) [inline src/main/java/controller/homepage/HomepageState.java:7]: Should be List<String>, and the actual obj is ArrayList

**Review by ethanliu24 — APPROVED** [review]

**Pull request merged**